ens-improvement-proposals/ensip-xx-ccip-metadata.md

```diff
@@ -21,52 +21,69 @@ This ENSIP addresses this by adding a way of important metadata to be gathered o
 
 ### Specification
 
-Add metadata functions to the resolver. The first argument MUST have the node if it is specifically related to a name (e.g. vitalik.eth). Most of these fields will be expected to be held off-chain via ccip-read.
+The metadata should include 2 different types of info
+
+- Offchain data storage location related info: `graphqlurl` includes the URL to fetch the metadata.
+
+- Ownership related info: `owner`, `isApprovedForAll` defines who can own or update the given record. The specification also must comply with [OwnedNode](https://github.com/corpus-io/Optimism-Resolver/blob/main/contracts/l2/L2PublicResolver.sol) that consists of the hash of the node and `msg.sender`. The OwnedNode is used when the resolver does not depend on the canonical registry nor a bridge to verify the ownership of the name on L1.
 
 ```solidity
 
 interface OffChainResolver {
     owner(bytes32 node) returns (bytes32); // first 20 bytes are assumed to be the address if it's an evm chain, but gives space for longer address on non-evm chains.
 
     isApprovedForAll(address account, address operator) returns (boolean);
-
-    dataLocation(bytes32 node) returns (string); 
-
-    allRecords(bytes32 node) returns (Records);
-
-    recordKeys(bytes32 node) returns (RecordsKeys)
-
+    graphqlurl() returns (string);
 }
-
 ```
 
-
-#### Example
-
 ```javascript
 const node = namehash('ccipreadsub.example.eth')
 const resolver = await ens.resolver(node)
-const owner = await resolver.owner(node) 
+const owner = await resolver.owner(node)
 // 0x123...
-const dataLocation = await.resolver.dataLocation(node)
+const dataLocation = await.resolver.graphqlurl()
 // {
-//   name: "Optimism",
-//   type: "Layer 2",
-//   chainId: 5   
+//   url: 'http://example.com/ens/graphql',
 // }
+```
 
-const recordKeys = await.resolver.recordKeys(node)
-// {
-//   coinTypes: [60, 1]
-//   texts: ['url', 'avatar']
-// }
+##### Ggraphql schema
+
+```graphql
+type Domain{
+  id: ID!
+  name: String
+  labelName: String
+  labelhash: Bytes
+  parent: Domain
+  subdomains: [Domain]
+  offchain: Offchain
+  resolver: (Resolver | OwnedResolver)
+}
+
+type Offchain(owner:){
+  chainId: ID!        # Id of the Chain (either ChainID or SLIP44 if non evm chain)
+  name: String        # Name of the Chain
+  isEVM: true/false   # True/False
```

Comment:
maybe it can be a more generic field? 
if the `vm` is not `evm` then it may be anything, which will require a special treatment. 

how about `vm` field as enum, union or object dictionary, so we can know exactly what type of `vm` will be handled. e.g. `evm / starkos / zkevm`